ui/views.py

```diff
@@ -188,6 +188,13 @@ def need_verified_email(request, *args, **kwargs):  # pylint: disable=unused-arg
     return standard_error_page(request, 401, "verify_email.html")
 
 
+def aouth_maintenance(request, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Returns maintenance page during oauth downtime
+    """
+    return standard_error_page(request, 503, "oauth_maintenance.html")
```

Comment:
I think this 503 status code might cause superfluous error reporting because it's a 5xx series. I agree 503 is probably the most accurate of the available statuses, I think this would be better as a 200, because it's an expected situation.

Reply:
Actually, I think a slightly better approach would be to add the maintenance page at it's own url (e.g. /login-offline) and in urls.py use `RedirectView` conditionally: https://docs.djangoproject.com/en/dev/ref/class-based-views/base/#redirectview